core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/SynapseExtension/SynapseExtensionUtilities.scala

```diff
@@ -0,0 +1,312 @@
+      s"""
+         |{
+         |  "displayName": "$runName-$now",
+         |  "description": "Synapse Spark Job Definition $artifactType",
+         |  "artifactType": "$artifactType"
+         |}
+         |""".stripMargin
+    val response = postRequest(ArtifactsUri, reqBody).asJsObject().convertTo[Artifact]
+    response.objectId
+  }
+
+  def createLakehouseArtifact(): String = {
+    val dtf = DateTimeFormatter.ofPattern("yyyyMMddHHmmss")
+    val now = dtf.format(LocalDateTime.now)
+
+    val reqBody: String =
+      s"""
+         |{
+         |  "displayName": "Lakehouse$now",
+         |  "description": "SynapseML Test Infra Lakehouse",
+         |  "artifactType": "Lakehouse"
+         |}
+         |""".stripMargin
+    val response = postRequest(ArtifactsUri, reqBody).asJsObject().convertTo[Artifact]
+    response.objectId
+  }
+
+  def deleteArtifact(artifactId: String): Unit = {
+    val uri: String = s"$SSPHost/metadata/artifacts/$artifactId"
+    deleteRequest(uri)
+  }
+
+  def submitJob(artifactId: String): String = {
+    val uri: String = s"$BaseUri/artifacts/$artifactId/jobs/sparkjob"
+    val createRequest = new HttpPost(uri)
+    createRequest.setHeader("Authorization", s"$AccessToken")
+
+    val response = postRequest(uri).asJsObject().convertTo[SparkJobDefinitionExecutionResponse]
+    response.artifactJobInstanceId
+  }
+
+  def monitorJob(artifactId: String, jobInstanceId: String): Future[String] = {
+    Future {
+      pollExecutionStatus(
+          artifactId, jobInstanceId,
+          SynapseUtilities.TimeoutInMillis,
+          System.currentTimeMillis())
+    }(ExecutionContext.global)
+  }
+
+  @tailrec
+  private def pollExecutionStatus(artifactId: String, jobInstanceId: String, timeout: Int, startTime: Long): String = {
+    val state = getJobStatus(artifactId, jobInstanceId)
+    if (state != null && state.statusString == "success") {
+      state.statusString
+    } else {
+      if ((System.currentTimeMillis() - startTime) > timeout) {
+        throw new TimeoutException(s"Job $jobInstanceId timed out.")
+      }
+      else if (state != null && Seq("Failed", "Dead", "Error").contains(state.statusString))
+      {
+        throw new RuntimeException(state.statusString)
+      }
+      else if (state  != null && state.statusString == "Completed")
+      {
+        state.statusString
+      }
+      else {
+        blocking {
+          Thread.sleep(8000)
+        }
+        pollExecutionStatus(artifactId, jobInstanceId, timeout, startTime)
+      }
+    }
+  }
+
+  def getJobStatus(artifactId: String, jobInstanceId: String): SparkJobDefinitionExecutionResponse = {
+    val uri = s"$BaseUri/artifacts/$artifactId/jobs/$jobInstanceId"
+    val response = getRequest(uri).asJsObject().convertTo[SparkJobDefinitionExecutionResponse]
+    println(s"$uri: $response")
+    response
+  }
+
+  val TimeoutMillis: Int = 100000
+
+  def postRequest(uri: String, requestBody: String  = ""): JsValue = {
```

Comment:
Can this be factored out from the 3 classes in some way? This also might apply more generally now that we have 3 examples of this kind of test

Reply:
Absolutely, this can be accomplished with an overall refactoring as discussed. I'm going to close this feedback for now. I'll fix the more blatant redundancy now and fix the rest after these get checked in